torch/distributed/checkpoint/state_dict.py

```diff
@@ -185,11 +186,48 @@ def _get_fqns(
                 fqn_obj_names.append(curr_obj_name)
         else:
             fqn_obj_names.append(curr_obj_name)
-            curr_obj = getattr(curr_obj, curr_obj_name)
+            if curr_obj_name == nn.modules.module._EXTRA_STATE_KEY_SUFFIX:
+                if i != len(obj_names) - 1:
+                    raise RuntimeError("Expect `_extra_state` to be the last obj name")
```

Comment:
Nit:
```suggestion
                    raise ValueError("Expect `_extra_state` to be the last obj name")
```

Reply:
This is not the API that receives an incorrect argument but the traverse logic is wrong. So ValueError doesn't seem to be more close to the meaning.